File: library/src/test/java/me/proxer/library/api/ucp/UcpTopTenEndpointTest.java
package me.proxer.library.api.ucp;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import me.proxer.library.entity.ucp.UcpTopTenEntry;
import me.proxer.library.enums.Category;
import me.proxer.library.enums.Medium;
import okhttp3.mockwebserver.MockResponse;
import org.junit.Test;

import java.io.IOException;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
public class UcpTopTenEndpointTest extends ProxerTest {

    @Test
    public void testDefault() throws ProxerException, IOException {
        server.enqueue(new MockResponse().setBody(fromResource("ucp_top_ten.json")));

        final List<UcpTopTenEntry> result = api.ucp()
                .topTen()
                .build()
                .execute();

        assertThat(result).first().isEqualTo(buildTestEntry());
    }

    @Test
    public void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("ucp_top_ten.json")));

        api.ucp().topTen()
                .build()
                .execute();

        assertThat(server.takeRequest().getPath()).isEqualTo("/api/v1/ucp/topten");
    }

    private UcpTopTenEntry buildTestEntry() {
        return new UcpTopTenEntry("640467", "2357", "High School DxD", Medium.ANIMESERIES,
                Category.ANIME);
    }
}

File: library/src/test/kotlin/me/proxer/library/api/ucp/UcpTopTenEndpointTest.kt
package me.proxer.library.api.ucp

import me.proxer.library.ProxerTest
import me.proxer.library.entity.ucp.UcpTopTenEntry
import me.proxer.library.enums.Category
import me.proxer.library.enums.Medium
import me.proxer.library.runRequest
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class UcpTopTenEndpointTest : ProxerTest() {

    private val expectedEntry = UcpTopTenEntry(
        id = "640467",
        entryId = "2357",
        name = "High School DxD",
        medium = Medium.ANIMESERIES,
        category = Category.ANIME
    )

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("ucp_top_ten.json") {
            api.ucp
                .topTen()
                .build()
                .safeExecute()
        }

        result.first() shouldBeEqualTo expectedEntry
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("ucp_top_ten.json") {
            api.ucp.topTen()
                .build()
                .execute()
        }

        request.path shouldBeEqualTo "/api/v1/ucp/topten"
    }
}
